### PR TITLE
Update/ref int tests

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Adds in integration tests to test complex types in Signal gen: Array types, Serializable Types, and nested Serializable and Array types.

### CODE REVIEW

File: Ref/test/int/ref_integration_test.py [hunk 1]
```diff
@@ -134,10 +143,38 @@ def test_send_command(self):
         self.assert_command("cmdDisp.CMD_NO_OP", max_delay=0.1)
         assert self.api.get_command_test_history().size() == 2
 
+    def test_send_command_args(self):
+        for count, value in enumerate(["Test String 1", "Some other string"], 1):
+            events = self.api.get_event_pred("cmdDisp.NoOpStringReceived", [value])
```
Comment (Joshua-Anderson) [inline Ref/test/int/ref_integration_test.py:148]: Are you intending to assert that this event is emitted? To my understanding this event object look unused.

File: Ref/test/int/ref_integration_test.py [hunk 2]
```diff
@@ -134,10 +143,38 @@ def test_send_command(self):
...
+                [
+                    value,
+                ],
+                max_delay=0.1,
+            )
+            assert self.api.get_command_test_history().size() == count
+
+    def test_send_command_args(self):
+        for count, value in enumerate(["Test String 1", "Some other string"], 1):
+            events = self.api.get_event_pred("cmdDisp.NoOpStringReceived", [value])
```
Comment (Joshua-Anderson) [inline Ref/test/int/ref_integration_test.py:160]: Same as above.

**Review by Joshua-Anderson — COMMENTED** [review]
Changes look great in general, glad to see a more advanced unit test example! Just had a single question.

**Comment by LeStarch** [comment]
@Joshua-Anderson ok, should be ready!

**Comment by ThibFrgsGmz** [comment]
@LeStarch What would you do if you wanted to run a test to verify the software version, CPU usage greater than 1% or memory usage greater than x KB? (that's what I want to do in https://github.com/nasa/fprime/discussions/1531)

**Comment by LeStarch** [comment]
Software version is publish in channels from SystemResources....so I'd do an assert on that telemetry value equal to the expected version (`git describe --tags --always`).   Is this enough to get you started, or do you want more detail?

**Pull request merged**